home/models.py

```diff
@@ -464,6 +467,18 @@ class Meta:
         indexes = [
             Index(fields=["status"])
         ]
+    
+    def get_content_with_course_links(self):
+        content = escape(self.content)
+        courses_replaced = []
+
+        for original_word in re.split(r' |\.', content):
+            word = re.sub(r'[\W_]+', '', original_word).upper()
+            if not word in courses_replaced and re.match(r'^([A-Z]{4}(?:[A-Z]|[0-9]){3,6})$', word) and Course.recent.filter(name=word).first():
```

Comment:
can we use a variable for this regex?

Reply:
@saewitz Added new commit which addresses this. Lmk what you think.